src/context/ProductContext.js

```diff
@@ -75,13 +84,17 @@ const ProductContextProvider = ({ children }) => {
   });
   useEffect(() => {
     (async () => {
-      const productData = await axios.get("/api/products");
-      productState.products.length === 0 &&
-        productDispatch({
-          type: "ADD_ALL_PRODUCTS",
-          payload: productData.data.products,
-        });
-      setProductsList(productData.data.products);
+      try {
+        const productData = await axios.get("/api/products");
+        productState.products.length === 0 &&
+          productDispatch({
+            type: "ADD_ALL_PRODUCTS",
+            payload: productData.data.products,
+          });
+        setProductsList(productData.data.products);
+      } catch (err) {
+        console.log(err);
```

Comment:
you can alert the error or create another state to show in UI

Reply:
Planning to do that in the future. Right now the focus was on priority features.